lib/compat/wordpress-6.6/rest-api.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * PHP and WordPress configuration compatibility functions for the Gutenberg
+ * editor plugin changes related to REST API.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers additional fields for search result rest api.
+ *
+ * @access private
+ * @internal
+ *
+ * @param  array $result_object      Search result object.
+ * @param  string $field_name        Current additional field's name (unused).
+ * @param  \WP_REST_Request $request Rest request object.
+ * @return string                    Human-readable label for the search result's post type.
+ */
+function _gutenberg_get_search_result_label_field( $result_object, $field_name, $request ) {
```

Comment:
Is it possible to put these functions into a directory/file structure that mirrors the intended Core target files? If it is then it will help when it comes to merge for WP 6.6.

If not then an `@core-merge` comment explaining how this code should be merged into WP Core for 6.6 would be super helpful.

Reply:
I've move the new media handler to the intended directory in core.

Regarding the addition of the new `label` property, I'm not sure how this should be handle. The changes touch existing classes in Core and we would need to copy the existing classes on the `wp-includes/rest-api/search` folder to do the changes. This seems a lot to me.

I kept the `register_rest_field` in gutenberg and I added a condition before to check if the field is present or not.

I've opened a PR in Core with the intended changes in the REST API link to this PR https://github.com/WordPress/wordpress-develop/pull/6112. I can move those changes in gutenberg if needed.

Reply:
I'm going to be unavailable for a few days but will check back in here as soon as I get a chance. Thank you for your work here.

Reply:
> I've opened a PR in Core with the intended changes in the REST API link to this PR https://github.com/WordPress/wordpress-develop/pull/6112. I can move those changes in gutenberg if needed.

The Core PR seems to contain other handlers as well and is still draft.

I think a good option would be to land the Media Search endpoint in Gutenberg first and allow it to stablise there whilst we solicit reviews from Core folks on your Core PR.

It's really great that you have the Core PR up early. Let's see if we can get some eyes on it. I'm going to ping @ellatrix who is Editor Tech Lead for WordPress 6.6 so she is aware.